lib/linux-list.ts

```diff
@@ -0,0 +1,115 @@
+import { PortInfo } from './binding-interface'
+
+import { spawn } from 'child_process'
+import Readline from '@serialport/parser-readline'
+
+// get only serial port names
+function checkPathOfDevice(path: string) {
+  return /(tty(S|WCH|ACM|USB|AMA|MFD|O|XRUSB)|rfcomm)/.test(path) && path
+}
+
+function propName(name: string) {
+  return {
+    DEVNAME: 'path',
+    ID_VENDOR_ENC: 'manufacturer',
+    ID_SERIAL_SHORT: 'serialNumber',
+    ID_VENDOR_ID: 'vendorId',
+    ID_MODEL_ID: 'productId',
+    DEVLINKS: 'pnpId',
+  }[name.toUpperCase()]
+}
+
+function decodeHexEscape(str: string) {
+  return str.replace(/\\x([a-fA-F0-9]{2})/g, (a, b) => {
+    return String.fromCharCode(parseInt(b, 16))
+  })
+}
+
+function propVal(name: string, val: string) {
+  if (name === 'pnpId') {
+    const match = val.match(/\/by-id\/([^\s]+)/)
+    return (match && match[1]) || undefined
```

Comment:
Thoughts on the [optional array access](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#array_item_access_with_optional_chaining)?

```ts
return match?.[1]
```